services/identity/DeploymentsService.ts

```diff
@@ -39,17 +39,18 @@ class DeploymentsService {
     site: Site
   }): Promise<Deployment> => {
     const amplifyResult = await this.createAmplifyAppOnAws(repoName)
-    const amplifyDomain = amplifyResult.map((result) => result.defaultDomain)
-
-    if (amplifyDomain.isErr()) {
-      logger.error(`Amplify set up error: ${amplifyResult}`)
-      throw amplifyDomain.error
+    if (amplifyResult.isErr()) {
+      logger.error(`Amplify set up error: ${amplifyResult.error}`)
+      throw amplifyResult.error
     }
+    const amplifyInfo = amplifyResult.value
```

Comment:
Note to self: do we not need .map?

Reply:
We don't need `.map` in this case. `.map` converts one kind of a result to another kind of result. In this case, I want to convert an Ok result to the value it contains (`amplifyResult.value`), and convert an Err result to a thrown error (`amplifyResult.error`). Since a result could contain one or the other (and never both), it does surprise me that you can call both `.value` and `.error` on the same variable. The type inference is smart enough to figure out when it is safe to call one or the other.

But I'll admit here that I find all of this kind of wonky. The code moves back and forth between two error handling approaches: one that throws Errors and another that returns errors in neverthrow results. Code like what you see here will appear wherever neverthrow results are processed by code that expects thrown Errors. This seems inevitable as long as we are mixing approaches.